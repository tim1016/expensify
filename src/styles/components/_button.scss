.button{
  background-color: $blue; 
  color: white;
  border: none;
  border-radius: .3rem;
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size $m-size;
}