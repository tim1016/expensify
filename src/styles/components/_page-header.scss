.page-header{
  background: #f7f7f7;
  margin: 0 0;
  padding: $l-size 0;
  &__title{
    margin: 0;
    font-weight: 300;
    span{
      font-weight: 600;
    }
  }
}