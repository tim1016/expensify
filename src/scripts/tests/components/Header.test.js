import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Header } from '../../components/Header';

test('Header shallow render', () => {
  const onClickSpy = jest.fn();
  const wrapper = shallow(<Header startLogout={onClickSpy}/>);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('Header Logout button spy', () => {
  const onClickSpy = jest.fn();
  const wrapper = shallow(<Header startLogout={onClickSpy} />);
  wrapper.find('button').simulate('click');
  expect(onClickSpy).toHaveBeenCalled();
});
