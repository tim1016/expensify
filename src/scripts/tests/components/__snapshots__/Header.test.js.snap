// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header shallow render 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
    Add expense
  </NavLink>
  <button
    onClick={[MockFunction]}
  >
    Logout 
  </button>
</header>
`;
